### PR TITLE
Add CLI backend

### PR DESCRIPTION
**Pull request by rukai**
A refactor I've been thinking about to eliminate the aws-sdk dependencies to save us from our terrible build times.
~~Still a long way way to go so I think I'll just work around the issue for now.~~
Ended up finishing it off.

The code quality is poor in some places as I'm hoping to delete this if the aws-sdk team can fix the build times, which they have promised to try

So on one hand this PR cuts out 5 minutes from an uncached release build.
But it also only cuts out 20 seconds from a cached release build and I was expecting to cut out a lot more (currently our cached release builds take 3 minutes)

I think it still makes sense to land at this point, although maybe I should have invested my time elsewhere in the first place.

### CODE REVIEW

File: aws-throwaway/src/backend/cli/instance_type.rs [hunk 1]
```diff
@@ -0,0 +1,3932 @@
...
+            "c6g.medium",
+            "c6g.metal",
+            "c6g.xlarge",
+            "c6gd.12xlarge",
+            "c6gd.16xlarge",
+            "c6gd.2xlarge",
+            "c6gd.4xlarge",
+            "c6gd.8xlarge",
+            "c6gd.large",
+            "c6gd.medium",
```
Comment (conorbros) [inline aws-throwaway/src/backend/cli/instance_type.rs:3202]: Did you pull this from a dump of all the instance types?

Comment (rukai) [reply aws-throwaway/src/backend/cli/instance_type.rs:3202]: yep, its the InstanceType copied directly from aws-sdk-ec2

**Review by conorbros — APPROVED** [review]

**Review by cjrolo — APPROVED** [review]
Approving this.

Why is build time so important for an end-user? I feel like adding extra code to support a build time a bit detrimental.

**Comment by rukai** [comment]
Thanks for the feedback. It actually really pained me to make this PR but I felt like I couldnt justify putting off the issue any longer.

AFAIK the only end users of this project is us, and we use it for running benchmarks in EC2, it doesnt end up in any user facing project.
For testing changes in benchmarks fast iterations is key and there are 2 cases this PR improves:
* Running the benchmark orchestrator from within a fresh EC2 instance.
    + we have no cache as its a fresh instance, it takes >10 minutes to compile just aws-sdk-ec2 on this instance while the rest of the project finishes in like ~5 mins. This PR cuts out the 10 minutes from aws-sdk-ec2, a huge time save.
    + this is currently a requirement when extracting shotover metrics from prometheus due to the way security groups are setup.
* Running the benchmark orchestrator locally which then generates EC2 instances.
    + Locally we can make use of our build cache, so builds take only 3 mins.
    + with this PR we improve our build time in this case by 20s. I'm hoping to find wins in other areas but 20s is a good start.
    
If the promised [aws-sdk-ec2 compile time improvements](https://github.com/awslabs/aws-sdk-rust/issues/113#issuecomment-1884902975) are good enough then we can delete this new CLI backend.
If they dont arrive after ~1 year then we should probably get rid of the SDK backend to reduce maintenance burden.

**Pull request merged**